### ISSUE TITLE
GFF source: ScoreType is required to parse the file but not required in GFF3 specs

### ISSUE THREAD

**Issue by terrymun** [intermine/intermine#1431]
Thanks for the amazing work on Intermine. I have recently started looking into publishing a mine for a model organism *Lotus japonicus*. I have followed the tutorial and everything checks out fine, with the exception of integrating the gff3 file using the default `gff` source (instead of `malaria-gff` in the tutorial):

```
BUILD FAILED
/var/www/intermine/imbuild/integrate.xml:54: The following error occurred while executing this line:
/var/www/intermine/imbuild/source.xml:253: java.lang.RuntimeException: value cannot be null for attribute Exon.scoreType
	at org.intermine.xml.full.Item.setAttribute(Item.java:353)
	at org.intermine.bio.dataconversion.GFF3Converter.process(GFF3Converter.java:360)
	at org.intermine.bio.dataconversion.GFF3Converter.parse(GFF3Converter.java:230)
	at org.intermine.bio.task.GFF3ConverterTask.execute(GFF3ConverterTask.java:241)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:440)
	at org.intermine.task.Integrate.performAction(Integrate.java:219)
	at org.intermine.task.Integrate.performAction(Integrate.java:131)
	at org.intermine.task.Integrate.execute(Integrate.java:123)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.Main.runBuild(Main.java:851)
	at org.apache.tools.ant.Main.startAnt(Main.java:235)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

I am suspecting that the gff3 description field might be parsed incorrectly, but frankly speaking I do not have an idea where to start. Perhaps I should write my own GFF parser, but I'm not very confident about it.

Here is an example gene from the gff3 file used for integration:

```
chr4	.	gene	3284600	3289068	.	.	.	ID=Lj4g3v0281040;Name=CUFF.46782;sequencetype=Protein coding
chr4	.	mRNA	3284600	3289068	.	+	.	ID=Lj4g3v0281040.1;Parent=Lj4g3v0281040;Name=Lj4g3v0281040.1;sequencetype=Protein coding;annotation=flagellin-sensing 2-like protein [Lotus japonicus] gi|355346200|gb|AER60531.1|
chr4	.	exon	3284600	3287825	.	+	.	ID=Lj4g3v0281040.1.exon.1;Parent=Lj4g3v0281040.1;sequencetype=Protein coding
chr4	.	exon	3288428	3289068	.	+	.	ID=Lj4g3v0281040.1.exon.2;Parent=Lj4g3v0281040.1;sequencetype=Protein coding
chr4	.	CDS	3284696	3287825	.	+	0	ID=Lj4g3v0281040.1.CDS.1;Parent=Lj4g3v0281040.1;sequencetype=Protein coding
chr4	.	CDS	3288428	3288774	.	+	2	ID=Lj4g3v0281040.1.CDS.2;Parent=Lj4g3v0281040.1;sequencetype=Protein coding
```

I'm sorry if this question sounds really amateur-ish, but it is my first time working with Intermine.

**Comment by julie-sullivan** [comment]
@terrymun Do any of your GFF files have "ScoreType=;" for exons? I am guessing our GFF parser is dumb and is not parsing the NULL values correctly. 

I could update the GFF parser to handle these in a non-dumb manner, or you could grep and remove those params. Let me know!

**Comment by terrymun** [comment]
Gave a shot in the dark and tried the `malaria-gff` parser, and the build is strangely successful. I will grep for the erroneous `scoreType` and see whether it comes up ;)

**Comment by terrymun** [comment]
@julie-sullivan No 'ScoreType=' string was found in the GFF3 file, which makes the case even more curious. I retract my previous comment of the build being successful—I was running it from an incorrect directory.

**Comment by julie-sullivan** [comment]
Can you send me (part of) your GFF file and I can test here?

**Comment by terrymun** [comment]
@julie-sullivan I have attached a very reduced version of the GFF file (zipped so that Github will host it)
, which after testing still throws the same error when building. There is a `sequencetype=` property in the attributes field, but removing it doesn't seem to make any difference.

The reduced GFF3 file: [20140314_Lj30_reduced.gff3.zip](https://github.com/intermine/intermine/files/482552/20140314_Lj30_reduced.gff3.zip)

Perhaps relevant: I am using the `gff` source for GFF3 parsing in my `project.xml`:

```xml
  <source name="lja-gff" type="gff">
      <property name="gff3.taxonId" value="34305"/>
      <property name="gff3.seqDataSourceName" value="Lotus japonicus"/>
      <property name="gff3.dataSourceName" value="Lotus japonicus"/>
      <property name="gff3.seqClsName" value="Chromosome"/>
      <property name="gff3.dataSetTitle" value="Lotus japonicus genome"/>
      <property name="src.data.dir" location="/var/www/lotusmine/gff3"/>
    </source>
```

**Comment by julie-sullivan** [comment]
Thanks! Trying now.

**Comment by julie-sullivan** [comment]
The second column in GFF is `source`, which InterMine uses as the `scoreType`. According to the file specifications, this is not a required field. However, an exception is thrown if these data are not provided. This is wrong obviously. Adding a null check.

**Comment by julie-sullivan** [comment]
@terrymun Yes, this was an InterMine bug! Here is my fix: dcc8370a4dc1d490ee65a01165d4779a545fcaec. Can you pull in that change (or just copy it over, just a couple lines) and let me know if that works?

**Comment by terrymun** [comment]
Hi @julie-sullivan, thanks for the very helpful advice and quick fix. I can confirm that the GFF3 file is being parsed properly now. Speaking of which, perhaps I should actually provide some kind of `source` in the second column of the GFF3 file... despite it being optional.

Thanks again for all the help, I sincerely appreciate it :) I can't wait to see how Intermine works out with *Lotus* data.
